onnxruntime/core/framework/sequential_executor.cc

```diff
@@ -241,22 +235,26 @@ Status SequentialExecutor::Execute(const SessionState& session_state, const std:
     utils::DumpNodeInputs(op_kernel_context, p_op_kernel->Node());
 #endif
 
+    const std::string node_name_for_profiling = [&]() -> std::string {
```

Comment:
regarding Scott's original comment:

This line could potentially add significant cost to a small model. If we're going to do this, shouldn't we do it when loading the model so it's not a per-Run overhead?

it seems like this is still being called per run

Reply:
per discussion with Sherlock, it's not trivial to do so, so we move most of the logic to profiling logic, except that the empty string will get recreated every time. The other way is to copy the same logic into 3 profiling blocks, but it looks a bit ugly.